integration_tests/src/main/python/hash_aggregate_test.py

```diff
@@ -624,18 +624,18 @@ def test_decimal128_min_max_group_by(data_gen):
             .agg(f.min('b'), f.max('b')))
 
 @ignore_order(local=True)
-@pytest.mark.parametrize('data_gen', [float_gen, double_gen], ids=idfn)
-def test_float_max_reduction_with_nan(data_gen):
+@pytest.mark.parametrize('data_gen', [SetValuesGen(t, [math.nan, None]) for t in [FloatType(), DoubleType()]], ids=idfn)
+def test_float_min_max_reduction_all_nans(data_gen):
     assert_gpu_and_cpu_are_equal_collect(
-        lambda spark: unary_op_df(spark, data_gen).selectExpr('max(a)'))
+        lambda spark: unary_op_df(spark, data_gen).selectExpr('min(a)', 'max(a)'))
 
 @ignore_order(local=True)
-@pytest.mark.parametrize('data_gen', [float_gen, double_gen], ids=idfn)
-def test_float_max_group_by_with_nan(data_gen):
+@pytest.mark.parametrize('data_gen', [float_gen, double_gen, SetValuesGen(FloatType(), [math.nan, None]), SetValuesGen(DoubleType(), [math.nan, None])], ids=idfn)
+def test_float_min_max_group_by_with_nan(data_gen):
```

Comment:
This is similar. Would it be better to just update the group by tests that use min and max to include float_gen and double gen.  Then we can also have some hard coded data to handle the different corner cases more like a unit test.

Reply:
updated.